src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultDisplayTemplates.cs

```diff
@@ -25,12 +26,12 @@ public static string BooleanTemplate(IHtmlHelper htmlHelper)
 
             return htmlHelper.ViewData.ModelMetadata.IsNullableValueType ?
                 BooleanTemplateDropDownList(htmlHelper, value) :
-                BooleanTemplateCheckbox(value ?? false);
+                BooleanTemplateCheckbox(value ?? false, htmlHelper.HtmlEncoder);
         }
 
-        private static string BooleanTemplateCheckbox(bool value)
+        private static string BooleanTemplateCheckbox(bool value, IHtmlEncoder htmlEncoder)
```

Comment:
This will work, but it doesn't seem great to thread this service through all of HtmlHelper's code. let's try and find a better way.